### PR TITLE
doc: Add index to documentation

### PR DESCRIPTION
**Pull request by prashanthpai**
and add usage of `retest this please`

Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,10 +9,7 @@ GlusterD, while also becoming more modular and easing extensibility.
 
 ## Documentation
 
-* [Quick Start User Guide](doc/quick-start-user-guide.md)
-* [Development Guide](doc/development-guide.md)
-* [Coding Guidelines](doc/coding.md)
-* [REST API Reference](doc/endpoints.md)
+Complete documentation for users and developers can be found [here](doc/index.md)
```
Comment (rishubhjain) [inline README.md]: IMO Its better to point out User Documentation and Developer Documentation inside Readme.md itself.

Comment (prashanthpai) [reply README.md]: Good point. Will change :+1:

Comment (prashanthpai) [reply README.md]: @rishubhjain Done

File: doc/testing.md [hunk 2]
```diff
@@ -48,3 +48,16 @@ tests.
...
+
+The CentOS CI system runs `e2e` tests for every proposed PR and reports the
+result back in the PR with a link to output of the tests.
+
+If you are certain that the tests run by CentOS CI have failed spuriously, you
+can retrigger running the `e2e` tests by writing the following magic phrase
+on the PR as a comment:
+
+```
+retest this please
```
[the remark this answers is not in the document]
Comment (prashanthpai) [reply doc/testing.md]: Anybody can trigger the retest of centos CI.

On a related note - For a first time contributor's PR, tests need to be manually triggered (only once) by maintainers with the phrase `add to whitelist`. That is intentionally not documented.

**Comment by prashanthpai** [comment]
retest this please

File: doc/testing.md [hunk 3]
```diff
@@ -48,3 +48,16 @@ tests.
 
 The argument to the -run command-line flag is an unanchored regular expression
 that matches the test's name.
+
+## Retriggering tests in the cnetos CI system
```
Comment (harigowtham) [inline doc/testing.md]: nit: centos

File: doc/testing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (harigowtham) [reply doc/testing.md]: I don't find us mentioning that the PR needs the centos CI to pass for it to get merged. Not sure if this is the right place or the review process section of the developement guide is the right section.

[the remark this answers is not in the document]
Comment (rishubhjain) [reply doc/testing.md]: I agree with @harigowtham , testing document should have details regarding testing code. Testing on github should be under a different section. @harigowtham As we are restructuring the documentation we can take up this task of finding the right place to put it as Prashanth has already provided with the right content.

[the remark this answers is not in the document]
Comment (prashanthpai) [reply doc/testing.md]: > Not sure if this is the right place or the review process section of the developement guide is the right section.

Development guide would be the right place. There's no harm in that being present here redundantly. Feel free to add this to development guide as well when you restructure

File: doc/testing.md (hunk 3 above)
Comment (prashanthpai) [reply doc/testing.md]: Done

**Pull request merged**